src/Core/Import/EntityField/Provider/EntityFieldsProviderFinder.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Import\EntityField\Provider;
+
+use PrestaShop\PrestaShop\Core\Import\Exception\NotSupportedImportEntityException;
+
+/**
+ * Class EntityFieldsProviderFinder defines an entity fields provider finder.
+ */
+final class EntityFieldsProviderFinder implements EntityFieldsProviderFinderInterface
```

Comment:
for me it seems more like some registry/store than a finder, because it does not try to find, but rather check if its registered, wdyt?

Reply:
Makes sense

Reply:
yeah I agree that `Finder` is not the best name, but not sure about how to actually name it :thinking: for me it's something close to a `Strategy`, but that's not very precise as well :thinking: registry/store are decent names, but maybe there are more options? calling @PierreRambaud for naming :smile: